modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -3469,6 +3456,8 @@ else if (log.isTraceEnabled())
                                         }
                                     }
 
+                                    updateLastSentMessageTime();
```

Comment:
what the reason to set sent time here?

Reply:
We’ve just initialized successful connection to next node. We just sent TcpDiscoveryHandshakeRequest message. This means: this connection is definitely ok and should not be checked right now. Also, we just sent a message. We rely on the time of last message sent and we fixate it here. Looks correct to me.